packages/store/addon/-private/system/record-array-manager.js

```diff
@@ -351,18 +351,17 @@ function updateLiveRecordArray(array, internalModels) {
 
 function removeFromAdapterPopulatedRecordArrays(internalModels) {
   for (let i = 0; i < internalModels.length; i++) {
-    let internalModel = internalModels[i];
-    let list = internalModel._recordArrays.list;
-
-    for (let j = 0; j < list.length; j++) {
-      // TODO: group by arrays, so we can batch remove
-      list[j]._removeInternalModels([internalModel]);
-    }
-
-    internalModel._recordArrays.clear();
+    removeFromAll(internalModels[i]);
   }
 }
 
+function removeFromAll(internalModel) {
+  internalModel._recordArrays.forEach(function(recordArray) {
```

Comment:
Why move to `.forEach` here instead of sticking with the original looping?

Reply:
good point. I moved to forEach at first trying to resolve the "creating an extra nested closure"  lint error (which didn't work), now that it is a helper function the original loop should be fine, will restore